generator/src/lib.rs

```diff
@@ -2407,6 +2407,7 @@ pub fn generate_struct(
         #dbg_str
         #[derive(Copy, Clone, #default_str #manual_derive_tokens)]
         #[doc = #khronos_link]
+        #[must_use = "Vulkan structs are moved by builders"]
```

Comment:
Thoughts on the message?

Reply:
I think talking about "moved" for `Copy` types is confusing. How about "Builders create new updated structs"?

Even then, referring to builders will be confusing when this inevitably comes up in other contexts. Maybe the best thing is no custom message?

Reply:
> Even then, referring to builders will be confusing when this inevitably comes up in other contexts.

That is why I was hesitant to move this to the `struct` definition in the first place; I haven't come across such a case yet but would love to know when it comes up and if the `must_use` makes sense there at all.

Your suggestion is much better though, I would even consider `Builders create new updated Vulkan structs`, maybe even add `... that must be consumed`. And then find different wording for `must`, since it's more of a suggestion that _not_ using the resulting object indicates an accidental no-op.

Reply:
> That is why I was hesitant to move this to the struct definition in the first place; I haven't come across such a case yet but would love to know when it comes up and if the must_use makes sense there at all.

I can think of lots of reasons why you might return a Vulkan struct from user code. For example:
- Abstracting out a common pattern
- Decoding serialized information into a Vulkan format (e.g. `VkSamplerCreateInfo`)
- Yielding memory barriers for accumulation elsewhere

 `must_use` is desirable in all of these cases, though it'll be a bit odd to see messages about builders for most of them, or discussion of "updated structs" when an API is creating the struct internally to begin with.

Incidentally, I think "builders" might be poor word choice regardless, since we've ditched the idea of specific objects that could be named as such. Maybe "helper functions"?

> I would even consider Builders create new updated Vulkan structs, maybe even add ... that must be consumed. And then find different wording for must, since it's more of a suggestion that not using the resulting object indicates an accidental no-op.

rustc already unconditionally says "... that must be used" here; I think that's sufficient.

Reply:
Apologies, I more so meant cases where `#[must_use]` could trigger but the user is thoughtfully ignoring the result. Maybe a function that performs an operation and returns a Vulkan struct that the user might optionally use for reference, but that seems particularly niche (e.g. `Option::get_or_insert_with()` style, where I might _not always_ want to use the returned value; but it fortunately returns a `&mut` anyway).

Will strip the message out, good points.

Reply:
Yeah, explicitly ignoring values in those corner cases makes sense.